### DISCUSSION TITLE
React 13 font

### DISCUSSION THREAD

**Discussion by guilhermewm** [vercel/next.js#47412] [Help] [answered]
### Summary

I would like to know if there is any major difference between using local fonts on the HEAD (https://nextjs.org/docs/basic-features/font-optimization#apply-the-font-in-head), or using it as className. There are any recommendations from the team? In terms of implementation, what is the best option?

**Comment by icyJoseph** [comment, marked as answer]
I think you mean Next 13's font module, `next/font`?

What this package does, besides making the font available, is giving you a classname you can use around your app, or it gives you the name of the font family it'll use. 

So if I use the recursive font, I'll actually have something like: 

![Screenshot 2023-03-23 at 11 30 26](https://user-images.githubusercontent.com/21013447/227176275-038e6cad-d12c-4f12-997e-18a12be82e3d.png)

While, using the class gives me access to a CSS Rule, that contains font related declarations.

![Screenshot 2023-03-23 at 11 32 29](https://user-images.githubusercontent.com/21013447/227176729-ef77fed1-e523-47c4-95a6-b0fe82f33177.png)

In short, there's no inherently bad or good approach here, but the ability to naturally use CSS. Either you want the font to apply globally, through inheritance, or you can scope it to the certain sections of the DOM tree, again relying on inheritance. In the case of Next.js' architecture, you can easily apply fonts at a page by page level.

I think the general recommendation is, to always be able to answer this question: Can you predict what your CSS will do?

Specifically for local fonts I don't know if there's anything more to consider. This article https://web.dev/font-best-practices/#using-self-hosted-fonts has some gotchas, which we solve by using the `next/font` package, so I think it should be about the same.
